### PR TITLE
added topography raster extraction utility class with tests

### PR DESCRIPTION
**Pull request by grantbuster**

### CODE REVIEW

File: sup3r/preprocessing/data_handling.py [hunk 1]
```diff
@@ -623,8 +624,11 @@ def time_chunk_size(self):
         if self._time_chunk_size is None:
             step_mem = self.feature_mem * len(self.raw_features)
             step_mem /= len(self.time_index)
-            self._time_chunk_size = np.min([np.int(1e9 / step_mem),
-                                            self.n_tsteps])
+            if step_mem == 0:
```
Comment (bnb32) [inline sup3r/preprocessing/data_handling.py:627]: this should probably just be n_tsteps, although it won't really matter.

Comment (grantbuster) [reply sup3r/preprocessing/data_handling.py:627]: step_mem -> n_tsteps?

Comment (grantbuster) [reply sup3r/preprocessing/data_handling.py:627]: or you're saying i should set `self._time_chunk_size = self.n_tsteps`?

Comment (bnb32) [reply sup3r/preprocessing/data_handling.py:627]: Yeah time_chunk_size = n_steps. It'll cap to that anyway with 1e9.

**Review by bnb32 — APPROVED** [review]
This looks good. I'm wondering you didn't use DataHandlerH5 to extract the source topo data and use hr_spatial_coarsen instead of using the nn aggregation though?

**Comment by grantbuster** [comment]
The sup3r data handler classes spend a lot of time doing rasterization which will fail when sourcing topo data from the full-disc NSRDB which is flattened and sparsified (for example). It makes more sense and is more flexible to use the rex.Resource handler and keep the elevation data flattened.

**Comment by bnb32** [comment]
> The sup3r data handler classes spend a lot of time doing rasterization which will fail when sourcing topo data from the full-disc NSRDB which is flattened and sparsified (for example). It makes more sense and is more flexible to use the rex.Resource handler and keep the elevation data flattened.

Gotcha, makes sense.

**Pull request merged**